Realm/RLMUtil.mm

```diff
@@ -436,7 +389,64 @@ BOOL RLMIsRunningInPlayground() {
     return [[NSBundle mainBundle].bundleIdentifier hasPrefix:@"com.apple.dt.playground."];
 }
 
-id RLMMixedToObjc(realm::Mixed const& mixed) {
+realm::Mixed RLMObjcToMixed(__unsafe_unretained id v, __unsafe_unretained RLMRealm *realm) {
+    if (!v || v == NSNull.null) {
+        return realm::Mixed();
+    }
+
+    RLMPropertyType type;
+    if ([v isKindOfClass:[RLMValueBase class]]) {
+        v = [v rlmValue];
+        type = [v valueType];
+    }
+    else if ([v conformsToProtocol:@protocol(RLMValue)]) {
+        type = [v valueType];
+    }
+    else {
+        REALM_TERMINATE("Unexpected Type");
+    }
+
+    switch (type) {
+        case RLMPropertyTypeInt:
+            return realm::Mixed([(NSNumber *)v intValue]);
+        case RLMPropertyTypeBool:
+            return realm::Mixed([(NSNumber *)v boolValue]);
+        case RLMPropertyTypeFloat:
+            return realm::Mixed([(NSNumber *)v floatValue]);
+        case RLMPropertyTypeDouble:
+            return realm::Mixed([(NSNumber *)v doubleValue]);
+        case RLMPropertyTypeUUID:
+            return realm::Mixed(RLMObjcToUUID(v));
+        case RLMPropertyTypeString:
+            return realm::Mixed([(NSString *)v cStringUsingEncoding:NSUTF8StringEncoding]);
+        case RLMPropertyTypeData:
+            return realm::Mixed(realm::BinaryData((const char*)[(NSData *)v bytes],
+                                                  (size_t)[(NSData *)v length]));
+        case RLMPropertyTypeDate:
+            return realm::Mixed(realm::Timestamp([(NSDate *)v timeIntervalSince1970], 0));
+        case RLMPropertyTypeObject: {
+            // If we are unboxing an object and it is unmanaged, we need to
```

Comment:
This is called in places where it absolutely should not be mutating anything. This function needs to take a CreatePolicy and honor that.

Reply:
Who should be in charge of the CreatePolicy? Should it be `ValueUpdater` in Object Store?

Reply:
unbox() needs to forward on the value it was passed. Queries should pass CreatePolicy::Skip. The accessor function should pass CreatePolicy::SetLink.